modules/fedimint-ln-client/src/lib.rs

```diff
@@ -702,6 +702,31 @@ enum PayError {
 }
 
 impl LightningClientModule {
+    // Ping gateway endpoint to verify that it is available before locking funds in
+    // OutgoingContract
+    async fn verify_gateway_availability(&self, gateway: &LightningGateway) -> anyhow::Result<()> {
+        let response = reqwest::Client::new()
+            .post(
+                gateway
+                    .api
+                    .join("is_available")
+                    .expect("is_available contains no invalid characters for a URL")
+                    .as_str(),
+            )
+            .json(&())
+            .send()
+            .await
+            .map_err(|e| anyhow::anyhow!("Gateway is not available: {e}"))?;
```

Comment:
nit: I think the more idiomatic way would be using `.context("Gateway is not available")?` here.